conn.go

```diff
@@ -222,6 +219,18 @@ func (c *Conn) authAllowed() bool {
 	return !c.server.AuthDisabled && (isTLS || c.server.AllowInsecureAuth)
 }
 
+// handleError writes errors responses and closes the connection once too many
+// have occurred.
+func (c *Conn) handleError(code int, ec EnhancedCode, msg string) {
```

Comment:
This name seems to imply it should be called for all errors. Perhaps it could be more specific? e.g. `protocolError`

Reply:
Sure, `protocolError` works for me.